runtime/near-wallet-contract/implementation/wallet-contract/src/near_action.rs

```diff
@@ -0,0 +1,87 @@
+//! Definition of `Action` for Near protocol.
+//! Unfortunately we cannot use `near-primitives` directly in the contract
+//! because it uses dependencies that do not compile to Wasm (at least
+//! not without some extra feature flags that `near-primitives` currently
+//! does not include).
+
+use near_sdk::{AccountId, Gas, NearToken, PublicKey};
+
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub enum Action {
```

Comment:
If that is ok I would only include only the actions we actually support by the Wallet Contract.
Recently we introduced a new action for nonrefundable-storage and I wonder how do we plan to maintain the list here.

Reply:
My idea with including all the actions was to make it cleat what actions are explicitly forbidden (as opposed to actions which may have been forgotten about). It is a good point about the maintainability though. Maybe I'll only include the ones which are used and include a comment that it is intentional some are left out.

Reply:
Unused actions removed in https://github.com/near/nearcore/pull/10850/commits/9bb90c09357d822db77d86637bf8e2ce61a7dbbc